dotnet/samples/KernelSyntaxExamples/Example14_SemanticMemory.cs

```diff
@@ -64,6 +89,61 @@ public static async Task RunAsync()
         await RunExampleAsync(kernelWithCustomDb);
     }
 
+    public static async Task RunLocalExampleAsync()
+    {
+        Console.WriteLine("====================================================");
+        Console.WriteLine("======== Semantic Memory (local embeddings) ========");
+        Console.WriteLine("====================================================");
+
+        /* You can also use a local embedding generator.
+         *
+         * In this example we use a local embedding generator that uses a local
+         * embedding generator based on Microsoft.ML tokenizers.
+         */
+
+        var kernelWithLocalEmbeddings = Kernel.Builder
+            .WithLoggerFactory(ConsoleLogger.LoggerFactory)
+            .WithMicrosoftMLTextEmbeddingGenerationService(serviceId: "roberta", setAsDefault: true) // TODO Files are missing
+            .WithMemoryStorage(new VolatileMemoryStore())
+            .Build();
+
+        await RunExampleAsync(kernelWithLocalEmbeddings);
+    }
+
+    public static async Task RunLocalCustomExampleAsync()
+    {
+        Console.WriteLine("====================================================");
+        Console.WriteLine("======== Semantic Memory (local custom embeddings) ========");
+        Console.WriteLine("====================================================");
+
+        /* You can also use a local embedding generator.
+         *
+         * In this example we use a local embedding generator that uses a local
+         * embedding generator based on Microsoft.ML tokenizers.
+         */
+
+        var encoder = EmbeddedResource.ReadStream("EnglishRoberta.encoder.json");
+        var vocab = EmbeddedResource.ReadStream("EnglishRoberta.vocab.bpe");
+        var dict = EmbeddedResource.ReadStream("EnglishRoberta.dict.txt");
+
+        if (encoder is null || vocab is null || dict is null)
+        {
+            throw new System.IO.FileNotFoundException("Missing required resources");
+        }
+
+        Microsoft.ML.Tokenizers.EnglishRoberta model = new(encoder, vocab, dict);
+
+        Microsoft.ML.Tokenizers.Tokenizer tokenizer = new(model, new Microsoft.ML.Tokenizers.RobertaPreTokenizer());
+
+        var kernelWithLocalEmbeddings = Kernel.Builder
+            .WithLoggerFactory(ConsoleLogger.LoggerFactory)
+            .WithMicrosoftMLTextEmbeddingGenerationService(tokenizer: tokenizer, serviceId: "roberta", setAsDefault: true) // TODO Array lengths must be equal
```

Comment:
TODO -- I believe something needs to be done to normalize the embeddings for cosine similarity comparison. Right now, these embeddings cannot be used. Maybe BPE is okay but could you please add some example files to use for that to the sample project with examples @weldpua2008 ? Thank you!